src/tests/system/tests/test_config.py

```diff
@@ -0,0 +1,175 @@
+""" SSSD Configuration-related Test Cases
+
+:requirement: IDM-SSSD-REQ: Configuration merging
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.topology import KnownTopologyGroup
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_config__change_config_while_sssd_running(client: Client):
+    """
+    :title: Re-generating config while SSSD is running
+    :setup:
+        1. In pam domain set "debug_level" to 9
+        2. Start SSSD
+    :steps:
+        1. Check that "debug_level" in pam domain is 9
+        2. Change "debug_level" in pam to 1
+        3. Apply config changes
+        4. Call "sssd --genconf"
+        5. Check that "debug_level" in pam is 1
+    :expectedresults:
+        1. "debug_level" is set to 9
+        2. "debug_level" is changed successfully
+        3. Changes are apllied successfully
+        4. "sssd --genconf" is called successfully
+        5. "debug_level" is set to 1
+    :customerscenario: False
+    """
+    client.sssd.pam["debug_level"] = "9"
+    client.sssd.start()
+
+    result = client.ldb.search("/var/lib/sss/db/config.ldb", "cn=pam,cn=config")
+    assert result["cn=pam,cn=config"]["debug_level"] == ["9"]
+
+    client.sssd.pam["debug_level"] = "1"
+    client.sssd.config_apply()
+    client.sssd.genconf()
+
+    result = client.ldb.search("/var/lib/sss/db/config.ldb", "cn=pam,cn=config")
+    assert result["cn=pam,cn=config"]["debug_level"] == ["1"]
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_config__genconf_particular_section(client: Client):
+    """
+    :title: Re-generating only particular section in config while SSSD is running
+    :setup:
+        1. In pam domain set "debug_level" to 9
+        2. In nss domain set "debug_level" to 9
+        3. Start SSSD
+    :steps:
+        1. Check that "debug_level" in pam domain is 9
+        2. Check that "debug_level" in nss domain is 9
+        3. Change "debug_level" in pam and in nss to 1
+        4. Apply config changes
+        5. Call "sssd --genconf-section==pam"
+        6. Check that "debug_level" in pam is 1
+        7. Check that "debug_level" in nss remained 9
+    :expectedresults:
+        1. "debug_level" is set to 9
+        2. "debug_level" is set to 9
+        3. "debug_level" is changed successfully
+        4. Changes are apllied successfully
+        5. "sssd --genconf-section==pam" is called successfully
+        6. "debug_level" in pam is 1
+        7. "debug_level" in nss remains 9
+    :customerscenario: False
+    """
+    client.sssd.pam["debug_level"] = "9"
+    client.sssd.nss["debug_level"] = "9"
+    client.sssd.start()
+
+    result = client.ldb.search("/var/lib/sss/db/config.ldb")
+    assert result["cn=pam,cn=config"]["debug_level"] == ["9"]
+
+    result = client.ldb.search("/var/lib/sss/db/config.ldb", "cn=nss,cn=config")
```

Comment:
There is no need for this call, `cn=nss,cn=config` is already part of result. Similar on other places, keep calls to ldb.search to minimum. It is enough to call it once before genconf, once after genconf.